var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -130,9 +136,14 @@ class Charliecloud(AutotoolsPackage):
     # See https://github.com/spack/spack/pull/16049.
     conflicts("platform=darwin", msg="This package does not build on macOS")
 
-    # Squashfuse support
-    depends_on("squashfuse@0.1.105:", when="+squashfuse")
-    depends_on("squashfs", type="run", when="+squashfuse")
+    # Squashfuse support. FIXME: the current variant and dependencies reflect
+    # Charliecloud's automatic mount/un-mounting requirements. A more manual
+    # approach with squashfuse could implemented in a different variant.
+    with when("+squashfuse"):
+        depends_on("libfuse@3:", type=("build", "run", "link"), when="@0.32:")
```

Comment:
As of 0.36 we also depend on `pkg-config` to find `libfuse`. `libfuse` also has this dependency in Spack. I don't know if the convention is for us to specify that dependency also.

As a further annoyance, 0.36 depends on `pkg-config` even without SquashFUSE (see hpc/charliecloud#1829). I hope to fix this in 0.37.

Reply:
Looks like `pkgconf` is picked up by virtual of `python3` being a dependency for our docs. I am able to install 0.36 without adding `pkgconfig` to this spec with and without `+squashfuse`.

```
$ spack find --variants --groups -d charliecloud
-- linux-centos7-haswell / gcc@4.8.5 ----------------------------
charliecloud@0.36~docs~squashfuse build_system=autotools
[...]
    python@3.11.7+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060
[...]
        pkgconf@1.9.5 build_system=autotools
[...]
charliecloud@0.36~docs+squashfuse build_system=autotools
[...]
    python@3.11.7+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060
[...]
        pkgconf@1.9.5 build_system=autotools
[...]
    squashfuse@0.5.0+lz4+lzo~min_size+shared+static+xz+zlib+zstd build_system=autotools
        lz4@1.9.4+pic build_system=makefile libs=shared,static
        lzo@2.10 build_system=autotools libs=shared,static
        zstd@1.5.5+programs build_system=makefile compression=none libs=shared,static
```